fendermint/app/src/app.rs

```diff
@@ -752,6 +760,18 @@ where
             .await
             .context("end failed")?;
 
+        // check for upgrades in the upgrade_scheduler
+        if let Some(upgrade) = self.upgrade_scheduler.get_upgrade(request.height) {
+            // there is an upgrade scheduled for this height, lets run the migration
+            let res = (upgrade.migration)(request.height);
```

Comment:
I am surprised that the migration doesn't get access to the `exec_state`. How else is it going to change the actor stuff. There are pending changes in there as well, so if it did something out of bounds, it would be overwritten in the commit, which sets the new state root.

Reply:
@aakoshh I know that the migration function signature is now what it should be, its just taking and receiving ints for the purpose of get things to compile and see fendermint calling a migration function at specified epoch